### PR TITLE
Cartofy

### PR DESCRIPTION
**Pull request by dgaubert**
- Drop support for `yarn.lock`
- Support `package-lock.json`
- Replace `mapnik-pool` by `generic-pool` and make it closer to `tilelive-mapnik` to be able to customize `pool-size` & `max-waiting-clients`
- Upgrade `eslint` 4.x to 5.x
- Remove unused functionality:
  - `.getIndexableDocs()`
  - `.getInfo()`
  - `pathname` is no longer supported
  - Event emitter interface
  - Do not gather `stats`
  - Stop checking `max-tile-size`
  - `.getRaster()`
  - `@mapbox/sphericalmercator`

### CODE REVIEW

**Comment by Algunenano** [comment]
I see you happily removing code :heart: 

AFAIK, we don't use the getRaster part, right? Removing that enables removing `@mapbox/sphericalmercator`.

Also, since you are removing yarn.lock, I'd update the README and recommend using npm instead.

File: lib/map-pool.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    return resolve(map);
+                });
+            } catch (err) {
+                return resolve(err);
+            }
+        });
+    };
+}
+
+function mapDestoryFunctor () {
```
Comment (simon-contreras-deel) [inline lib/map-pool.js]: Typo: `mapDestroyFunction` and the usage

**Review by simon-contreras-deel — APPROVED** [review]
There are many changes to be aware of everything, but it seems ok.

File: lib/map-pool.js (hunk 1 above)
Comment (dgaubert) [reply lib/map-pool.js]: https://en.wikipedia.org/wiki/Functor

It might the concept is not applied in the right way though.

Comment (simon-contreras-deel) [reply lib/map-pool.js]: I didn't know the concept, thanks! 

And the other word? `destory` --> `destroy`

Comment (dgaubert) [reply lib/map-pool.js]: hahaha! f***ing dyslexia!

**Pull request merged**